src/app.py

```diff
@@ -936,37 +939,45 @@ def submit_dataset(uuid):
-        response = requests.put(put_url, json = dataset_request, headers = {'Authorization': 'Bearer ' + token, 'X-Hubmap-Application':'ingest-api' }, verify = False)
-        if not response.status_code == 200:
-            logger.error(f"call to {put_url} failed with code:{response.status_code} message:" + response.text)
-            return Response(response.text, response.status_code)
-        updated_dataset = response.json()
-        return jsonify(updated_dataset)
-    except HTTPException as hte:
-        return Response(hte.get_description(), hte.get_status_code())
     except Exception as e:
         logger.error(e, exc_info=True)
-        return Response("Unexpected error while creating a dataset: " + str(e) + "  Check the logs", 500)        
- 
+        return Response("Unexpected error while creating a dataset: " + str(e) + "  Check the logs", 500)
+    dataset_request['status'] = 'Processing'
+    put_url = commons_file_helper.ensureTrailingSlashURL(app.config['ENTITY_WEBSERVICE_URL']) + 'entities/' + uuid
+    response = requests.put(put_url, json=dataset_request,
+                            headers={'Authorization': 'Bearer ' + token, 'X-Hubmap-Application': 'ingest-api'},
+                            verify=False)
+    if not response.status_code == 200:
+        error_msg = f"call to {put_url} failed with code:{response.status_code} message:" + response.text
+        logger.error(error_msg)
+        return Response(str(error_msg), 500)
+    def call_airflow():
+        try:
+            r = requests.post(pipeline_url, json={"submission_id" : "{uuid}".format(uuid=uuid), "process" : app.config['INGEST_PIPELINE_DEFAULT_PROCESS'],"full_path": ingest_helper.get_dataset_directory_absolute_path(dataset_request, group_uuid, uuid),"provider": "{group_name}".format(group_name=AuthHelper.getGroupDisplayName(group_uuid))}, headers={'Content-Type':'application/json', 'Authorization': 'Bearer {token}'.format(token=AuthHelper.instance().getProcessSecret() )}, verify=False)
+            if r.ok == True:
+                """expect data like this:
+                {"ingest_id": "abc123", "run_id": "run_657-xyz", "overall_file_count": "99", "top_folder_contents": "["IMS", "processed_microscopy","raw_microscopy","VAN0001-RK-1-spatial_meta.txt"]"}
+                """
+                data = json.loads(r.content.decode())
+                submission_data = data['response']
+                dataset_request[HubmapConst.DATASET_INGEST_ID_ATTRIBUTE] = submission_data['ingest_id']
+                dataset_request[HubmapConst.DATASET_RUN_ID] = submission_data['run_id']
+            else:
+                error_message = 'Failed call to AirFlow HTTP Response: ' + str(r.status_code) + ' msg: ' + str(r.text)
+                logger.error(error_message)
+                dataset_request['status'] = 'Error'
+                dataset_request['pipeline_message'] = error_message
+                response = requests.put(put_url, json=dataset_request,
```

Comment:
@DerekFurstPitt You should check for a valid response from AirFlow and log an error to the log with the details if it doesn't return a 2xx response.

Reply:
At present, if airflow returns a status code 400 or greater, it does this 

```
error_message = 'Failed call to AirFlow HTTP Response: ' + str(r.status_code) + ' msg: ' + str(r.text)
logger.error(error_message)
```
per the else block beneath `if r.ok == True`. I believe r.ok is true only for status codes between 1-399 correct? Do you want a separate log for if its 301-399 or does that log work.

Reply:
@shirey I'm noticing a problem in this part of the script. Since you had me move the PUT call to entity before the subfunction that calls airflow (setting status to processing) and only had it called again if airflow call fails to change the status to `error`, that means a couple other fields we set in the event of a 200 would not be set. We do the following if the airflow status is 200: 

```submission_data = data['response']
   dataset_request['ingest_id'] = submission_data['ingest_id']
   dataset_request['run_id'] = submission_data['run_id']
```

I would move these fields above before we call entity at line 947, but I get the data from the airflow response. 

So it seems to me we either need to move the entity call back down to be inside of the subfunction, or we need to make another call to entity. Obviously the second choice seems less than ideal.